livereveal/main.js

```diff
@@ -34,177 +34,88 @@ if(IPython.version.substring(0, 1) === '2') {
-      cell.metadata.internals = {};
-      cell.metadata.internals.slide_type = '-';
-    }
-  }
-}
-
-function labelCells(){
-  var cells = IPython.notebook.get_cells();
-  for(var i=1; i< cells.length; i++){
-    var cell = cells[i];
-    var index = i - 1;
-    if (cell.metadata.slideshow.slide_type == 'slide') {
-      cells[i].metadata.internals.slide_type = 'slide';
-      cells[index].metadata.slide_helper = 'slide_end';
-      cells[index].metadata.internals.slide_helper = 'slide_end';
-    }
-    else if (cell.metadata.slideshow.slide_type == 'subslide'){
-      cells[i].metadata.internals.slide_type = 'subslide';
-      cells[index].metadata.slide_helper = 'subslide_end';
-      cells[index].metadata.internals.slide_helper = 'subslide_end';
-    }
-    else if (cell.metadata.slideshow.slide_type == 'fragment'){
-      cells[i].metadata.internals.slide_type = undefined;
-      cells[index].metadata.slide_helper = undefined;
-      cells[index].metadata.internals.slide_helper = undefined;
-      cells[i].metadata.internals.frag_number = i;
-      var j=1;
-      while (j < cells.length - i) {
-        cells[i + j].metadata.internals.frag_helper = 'fragment_end';
-        cells[i + j].metadata.internals.frag_number = i;
-        j++;
-      }
-    }
-    else if (cell.metadata.slideshow.slide_type == 'notes'){
-      cells[i].metadata.internals.slide_type = undefined;
-      cells[index].metadata.slide_helper = undefined;
-      cells[index].metadata.internals.slide_helper = undefined;
-    }
-    else if (cell.metadata.slideshow.slide_type == 'skip'){
-      cells[i].metadata.internals.slide_type = undefined;
-      cells[index].metadata.slide_helper = undefined;
-      cells[index].metadata.internals.slide_helper = undefined;
-    }
-    else if (cell.metadata.slideshow.slide_type == '-'){
-      cells[i].metadata.internals.slide_type = undefined;
-      cells[index].metadata.slide_helper = undefined;
-      cells[index].metadata.internals.slide_helper = undefined;
-    }
-  }
-
-  // The first cell that isn't explicitly skipped or notes needs to be the
-  // start of a slide.
-  var slide_type;
-  for (i=0; i < cells.length; i++) {
-      slide_type = cells[i].metadata.slideshow.slide_type;
-      if (slide_type !== 'skip' && slide_type !== 'notes') {
-        cells[i].metadata.slideshow.slide_type = 'slide';
-        cells[i].metadata.internals.slide_type = 'slide';
-        break;
-      }
-  }
-  // The last cell needs to be the end of a slide
-  cells[cells.length - 1].metadata.slide_helper = "slide_end";
-  cells[cells.length - 1].metadata.internals.slide_helper = "slide_end";
-
-}
-
-function labelIntraSlides(){
-  var cells = IPython.notebook.get_cells();
-  for(var i in cells){
-    var cell = cells[i];
-    if (cell.metadata.slideshow.slide_type == 'fragment') {
-      $('.cell:nth('+i+')').addClass("fragment");
-      $('.cell:nth('+i+')').attr('data-fragment-index', cell.metadata.internals.frag_number);
```

Comment:
Ah, gotcha. I think I've achieved the same thing in a different way. Given a 'fragment' cell followed by a '-' cell, your code was making something like this:

```html
<div class="fragment" data-fragment-index="1">cell 1</div>
<div class="fragment" data-fragment-index="1">cell 2</div>
```

My code produces something like this:

```html
<div class='fragment'>
  <div>cell1</div>
  <div>cell2</div>
</div>
```

i.e. it nests the cells of a single fragment in another container so they appear together. This was what I meant when I said that I needed to do something more on fragments - you can see the change I made to fix it [here](https://github.com/takluyver/live_reveal/commit/65a33bc37c0c00f32c09241610ec1ed5c3781557).